piqture/data_loader/mnist_data_loader.py

```diff
@@ -13,123 +13,130 @@
+
+    # Validate inputs
     if not isinstance(img_size, (int, tuple)):
-        raise TypeError(
-            "the input img_size must be of the type int or tuple[int, int]."
-        )
+        raise TypeError("img_size must be an int or tuple[int, int].")
 
     if isinstance(img_size, tuple) and not all(
         isinstance(size, int) for size in img_size
     ):
-        raise TypeError("the input img_size must be of the type tuple[int, int].")
-
-    # Check if batch_size is an int.
-    if batch_size:
-        if not isinstance(batch_size, int):
-            raise TypeError("The input batch_size must be of the type int.")
-
-    # Check if labels are a list.
-    if labels:
-        if not isinstance(labels, list):
-            raise TypeError("The input labels must be of the type list.")
-
-    if normalize_max and normalize_min:
-        # Define a custom mnist transforms.
-        mnist_transform = torchvision.transforms.Compose(
-            [
-                torchvision.transforms.ToTensor(),
-                torchvision.transforms.Resize(img_size),
-                MinMaxNormalization(normalize_min, normalize_max),
-            ]
-        )
-    else:
-        # When normalization is not requested; when normalize_min and max are None.
-        mnist_transform = torchvision.transforms.Compose(
-            [
-                torchvision.transforms.ToTensor(),
-                torchvision.transforms.Resize(img_size),
-            ]
-        )
-
-    # Partial application of custom collate function that calls
-    # collate_fn with args.
-    new_batch = []
-    custom_collate = partial(collate_fn, labels=labels, new_batch=new_batch)
+        raise TypeError("img_size tuple must contain integers.")
+
+    if not isinstance(batch_size_train, int) or not isinstance(batch_size_test, int):
+        raise TypeError("batch_size_train and batch_size_test must be integers.")
+
+    if labels is not None and not isinstance(labels, list):
+        raise TypeError("labels must be a list.")
+
+    if load not in {"train", "test", "both"}:
+        raise ValueError('load must be one of "train", "test", or "both".')
+
+    # Define the transform
+    mnist_transform = torchvision.transforms.Compose(
+        [
+            torchvision.transforms.ToTensor(),
+            torchvision.transforms.Resize(img_size),
+            (
+                MinMaxNormalization(normalize_min, normalize_max)
+                if normalize_min is not None
+                and normalize_max is not None  # pylint: disable=C0301
+                else torchvision.transforms.Lambda(lambda x: x)
+            ),
+        ]
+    )
 
-    # Download dataset.
-    mnist_train = datasets.MNIST(
+    # Load the full MNIST dataset
+    mnist_full = datasets.MNIST(
         root="data/mnist_data",
-        train=True,
+        train=True,  # Always load train to split later
         download=True,
         transform=mnist_transform,
     )
```

Comment:
PyTorch suggests to load train and test datasets separately. I would prefer keeping it the original way. 

```python
# Download dataset.
mnist_train = datasets.MNIST(
    root="data/mnist_data",
    train=True,
    download=True,
    transform=mnist_transform,
)

mnist_test = datasets.MNIST(
    root="data/mnist_data", train=False, download=True, transform=mnist_transform
)
```